20-red-knob-sea-star/api_definitions/rest/claims_repository.yaml

```diff
@@ -0,0 +1,367 @@
+openapi: 3.0.3
+
+info:
+  title: Claims Repository API
+  version: 0.1.0
+  description: REST API for managing the GHGA claims repository
+  license:
+    name: Apache 2.0
+    url: http://www.apache.org/licenses/
+  contact: 
+    url: https://www.ghga.de/about-us/contact
+
+servers:
+  - url: https://api.server.test/v1
+
+tags:
+  - name: claims
+    description: USer Claims
+
+paths:
+
+  /users/{user_id}/claims:
+
+    get:
+      operationId: get_claims
+      summary: Get all claims for a given user
+      description: Endpoint used to get all claims for a specified user.
+      tags:
+        - claims
+      parameters:
+        - name: user_id
+          in: path
+          required: true
+          schema:
+            title: Internal ID of the user
+            type: string
+      responses:
+        '200':
+          description: User claims have been retrieved.
+          content:
+            application/json:
+              schema:
+                title: Response 200 Get Claims Users  User Id  Claims Get
+                type: array
+                items:
+                  $ref: '#/components/schemas/Claim'
+        '401':
+          description: Not authorized to get user claims.
+        '404':
+          description: The user was not found.
+        '422':
+          description: Validation error in submitted user ID.
+
+    post:
+      operationId: post_claim
+      summary: Store a user claim
+      description: Endpoint used to store a new claim about a user.
+      tags:
+        - claims
+      parameters:
+        - name: user_id
+          in: path
+          required: true
+          schema:
+            title: Internal ID of the user
+            type: string
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ClaimData'
+        required: true
+      responses:
+        '201':
+          description: Claim was successfully stored.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Claim'
+        '400':
+          description: Claim cannot be stored.
+        '409':
+          description: Claim was already stored.
+        '422':
+          description: Validation error in submitted ID or claims data.
+
+    delete:
+      operationId: delete_claims
+      summary: Delete all claims about a given user
+      description: Endpoint used to delete all claims about a given user.
+      tags:
+        - claims
+      parameters:
+        - name: user_id
+          in: path
+          required: true
+          schema:
+            title: Internal user ID
+            type: string
+      responses:
+        '204':
+          description: User claims were successfully deleted.
+        '404':
+          description: The user was not found.
+        '422':
+          description: Validation error in submitted user ID.
```

Comment:
Should claims ever be deleted? Or just let them expire or be revoked?

Reply:
If we need to delete them, do we have a use case for this "delete all" endpoint or would the single delete endpoint be enough?

Reply:
May be useful if a user is deleted from the database. But of course we can also get all claims for the user and then remove them individually.

Reply:
Claims can conain personal data about the user, so we may not have a legal basis to store them forever.

Reply:
Actually the user claims could also be deleted automatically if the user is deleted (via events or database cleanup jobs), so we don't really need that.

The deletion of individual claims may be needed to correct mistakes. I think the instance that is able to create a claim should also be able to delete it. Proper auditing of such operations is a separate concern that can be handled elsewhere.

Reply:
So I have removed the endpoint to delete all claims for the user now, but I would like to keep the other one.